File: api/src/main/java/com/forum/Api/business/abstracts/PostService.java
package com.forum.Api.business.abstracts;

import com.forum.Api.business.requests.Post.CreatePostRequest;
import com.forum.Api.business.requests.Post.DeletePostRequest;
import com.forum.Api.business.requests.Post.UpdatePostRequest;
import com.forum.Api.business.responses.Post.GetAllPostResponse;
import com.forum.Api.business.responses.Post.GetPostsByCategoryResponse;
import com.forum.Api.entities.concretes.Post;

import java.util.List;

public interface PostService {
    List<GetAllPostResponse> getAll();
    List<GetPostsByCategoryResponse> getPostsByCategoryId(int categoryId, int skip, int limit);
    Post getPostById(int id);
    void createPost(CreatePostRequest createPostRequest);
    void deletePost(DeletePostRequest deletePostRequest);
    void updatePost(UpdatePostRequest updatePostRequest);
}

File: frontend/src/services/postService.js
import api from './api';

export const updatePost = async (postData) => {
    try {
        const response = await api.put('/posts/update', postData);
        return response;
    } catch (error) {
        throw error;
    }
}

export const addPost = async (postData) => {
    try {
        const response = await api.post('/posts/add', postData)
        return response;
    } catch (error) {
        throw error;
    }
}

export const deletePost = async (postId) => {
    try {
        const response = await api.delete(`posts/delete/${postId}}`)
        return response;
    } catch (error) {
        throw error;
    }
}

export const getAllPost = async () => {
    try {
        const response = await api.get('/posts/getall');
        return response;
    } catch (error) {
        throw error;
    }
};

export const getPostById = async (id) => {
    try {
        const response = await api.get(`/posts/${id}`);
        return response;
    } catch (error) {
        throw error;
    }
}

export const getPostByCategoryId = async (categoryId, skip, limit) => {
    try {
        const response = await api.get(`/posts/category/${categoryId}?skip=${skip}&limit=${limit}`);
        return response;
    } catch (error) {
        throw error;
    }
}